blog/_posts/2021-10-21-scala-3.1.0-released.md

```diff
@@ -0,0 +1,185 @@
+---
+layout: blog-detail
+post-type: blog
+by: Paweł Marks, VirtusLab
+title: Scala 3.1.0 released!
+---
+Hello from the Scala 3 team! It has already been six weeks since we have announced the release candidate for the first minor version after the initial release of Scala 3. Now, after two more RCs, we can confidently promote Scala 3.1.0-RC3 to a new stable release of the language. In other words, Scala 3.1.0 is officially out!
+
+## Compatibility notice
+
+This is a first *minor* release after the initial release of Scala 3.0. This has following consequences:
+
+- Scala 3.1 is backward binary compatible: you can use dependencies compiled with Scala 3.0 in 3.1 projects.
+- Scala 3.1 is _not_ forward binary compatible: you _cannot_ use dependencies compiled with Scala 3.1 in Scala 3.0 projects.
+- Updating Scala to 3.1 is not a forward compatible change. If you are a library maintainer and are following SemVer you may want to update in your next minor release.
+
+Although we cannot guarantee full source compatibility between minor versions, we have put a lot of effort into assuring that all code that was working in 3.0.2, except in some rare cases, will also work in 3.1.0. This means that if you are an application developer, you can confidently update the compiler version to take advantage of the newest improvements. You will still be able to use dependencies compiled with Scala 3.0.
+
+If you are a library maintainer, *updating to 3.1.0 will force all of your users to update to 3.1.0 as well*. We understand that the current state of binary compatibility may be unsatisfactory. We are actively working on technical solutions to support forward-compatibility in 3.2.0. In the meantime, we recommend testing your library with Scala 3.1.0 and 3.0.2, but _publishing_ it with 3.0.2. This will allow downstream users to use your library even if they can’t update to Scala 3.1.0. You can find an example of build that follows this recommendation [here](https://github.com/typelevel/scalacheck/pull/847).
```

Comment:
> We are actively working on technical solutions to support forward-compatibility in 3.2.0.

Is this a promise? It isn't, right? We aren't sure if this will actually happen in 3.2.0? If so, consider weakening the wording a little, perhaps to "We are exploring possible technical solutions to..."